### DISCUSSION TITLE
Add new social & board problem

### DISCUSSION THREAD

**Discussion by jyu99** [zephyrproject-rtos/zephyr#50734] [Q&A]
Hi 

I'm trying to add new Soc & board 

below is the problem When I use the west build the sample code hello world 

`[2/134] Generating include/generated/version.h
-- Zephyr version: 3.2.0-rc1 (/Users/pswd/zephyrproject/zephyr), build: v3.2.0-rc1-57-gb27c5d73ef09
[8/134] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj 
ccache /Users/pswd/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I/Users/pswd/zephyrproject/zephyr/kernel/include -I/Users/pswd/zephyrp
In file included from /Users/pswd/zephyrproject/zephyr/include/zephyr/arch/arm/aarch32/arch.h:20,
                 from /Users/pswd/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from /Users/pswd/zephyrproject/zephyr/include/zephyr/kernel_includes.h:33,
                 from /Users/pswd/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /Users/pswd/zephyrproject/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /Users/pswd/zephyrproject/zephyr/arch/arm/core/offsets/offsets.c:9:
/Users/pswd/zephyrproject/zephyr/include/zephyr/arch/arm/aarch32/asm_inline_gcc.h: In function 'arch_irq_lock':
/Users/pswd/zephyrproject/zephyr/include/zephyr/devicetree.h:326:40: error: 'DT_N_INST_0_arm_v7m_nvic_P_arm_num_irq_priority_bits' undeclared (first use in this function)
  326 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
`

`/Users/pswd/zephyrproject/zephyr/include/zephyr/devicetree.h:326:40: error: 'DT_N_INST_0_arm_v7m_nvic_P_arm_num_irq_priority_bits' undeclared (first use in this function)
  326 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))`

I wonder know which file should need to re-write or reconfigured

**Comment by erwango** [comment]
You likely miss the following in your SoC dts description:
```
&nvic {
	arm,num-irq-priority-bits = <4>;
};
```
To be tuned accordingly.

**Reply by jyu99** [reply]
here is my dits
/* SPDX-License-Identifier: Apache-2.0 */
/dts-v1/;

#include <arm/armv6-m.dtsi>
/ {


	chosen {

		zephyr,sram = &sram0;
		zephyr,flash = &flash0;
	};

	cpus {
		#address-cells = <1>;
		#size-cells = <0>;

		cpu@0 {
			compatible = "arm,cortex-m0";
			reg = <0>;
		};
	};

	sram0: memory@20000000 {
		compatible = "mmio-sram";
		reg = <0x20000000 0x20000>;
	};

	flash0: flash@10000000 {
		compatible = "soc-nv-flash";
		reg = <0x10000000 0x3D000>;

	};



};

&nvic {
	arm,num-irq-priority-bits = <2>;
};
